m2m-transfers/m2m-transfers-json-contract.md

```diff
@@ -0,0 +1,196 @@
+#JSON contract for BANNO.M2MTRANSFERS.V1.POW
+
+## GET PRELOADDATA STATE
+### Client Request (PRELOADDATA)
+
+```json
+{
+  "rgState": "PRELOADDATA",
+  "powerOnFileName": "BANNO.M2MTRANSFERS.V1.POW",
+  "userChrList": [
+    { "id": 1, "value": "1234567890" }, // 10-digit account
+    { "id": 2, "value": "" }, // userchar[2-5] is unused
+    { "id": 3, "value": "" },
+    { "id": 4, "value": "" },
+    { "id": 5, "value": "" }
+  ],
+  "userNumList": [], //  usernum[1-5] is unused
+  "rgSession": 1
+}
+```
+
+### PowerOn Successful Response (PRELOADDATA)
+list of institution and member limits, eligible shares, list of scheduled transfers, list of saved recipients
+
+```json
+{
+	"currentState": {
+		"enforceLimits": true,
+		"transferLimits": [{
```

Comment:
@tkainz could you send us whatever limit applies, and what the user has remaining of that limit?

Reply:
Instead of the current limits then and the user's current counts, you want to ALSO add the remaining amounts OR replace the current levels with the remaining amounts?
Currently:
	"currentState": {
		"enforceLimits": true,
		"transferLimits": [{
			"institutionCount": "5",
			"institutionAmount": "10000.00",
			"memberLimitCount": "3",
			"memberLimitAmount": "5000.00",
			"memberActualCount": "1",
			"memberActualAmount": "500.00"

NEW??
	"currentState": {
		"enforceLimits": true,
		"transferLimits": [{
			"institutionCount": "5",
			"institutionAmount": "10000.00",
			"memberLimitCount": "3",
			"memberLimitAmount": "5000.00",
			**"memberCountRemaining": "1",
			"memberAmountRemaining": "500.00"**

Reply:
@jrobinson01 do we need `"transferLimits": [{
"institutionCount": "5",
"institutionAmount": "10000.00",
"memberLimitCount": "3",
"memberLimitAmount": "5000.00",`

Reply:
If possible, I'd like just the limits we need to apply for the current user. I don't think the UX will have enough info about the configuration to decide which to apply. I think something like this would be ideal:
```javascript
"transferLimits": {
    "applyLimits": true,
    "count": "1",
    "amount":"1.00",
    "remaining": "1",
    "remainingAmount":"1.00" 
  }
```